build/Dockerfile.bcc.kepler

```diff
@@ -9,14 +9,22 @@ ENV GOPATH=/opt/app-root GO111MODULE=on GOROOT=/usr/local/go
 
 ENV PATH=$GOPATH/bin:$GOROOT/bin:$PATH
 
+# Set GOCACHE environment variable for caching
+ENV GOCACHE=/go/cache 
+
 WORKDIR $GOPATH/src/github.com/sustainable-computing-io/kepler
 
+# Cache and download Go dependencies if go.mod or go.sum changes
+COPY go.mod go.sum ./
+RUN --mount=type=cache,target=/go/cache \
+go mod download -x
 
 COPY . .
 RUN mkdir -p data
 
-# Build kepler
-RUN make build SOURCE_GIT_TAG=$SOURCE_GIT_TAG BIN_TIMESTAMP=$BIN_TIMESTAMP
+# Build kepler with caching to speed up the process
+RUN --mount=type=cache,target=/go/cache/ \
+make build SOURCE_GIT_TAG=$SOURCE_GIT_TAG BIN_TIMESTAMP=$BIN_TIMESTAMP
```

Comment:
which is interested to me as https://github.com/sustainable-computing-io/kepler/actions/runs/6878164112/job/18707236187#step:5:980 shows the logic to me.
see this:
as we copied everything from kepler repo into build image at line 22, which means we overwrite line 18. the go.mod and go.sum. and then there is a go package download process happened.
imo, it's worth from this pr to try with `go mod download -x` before `go mod tidy` to load go mod from vendor folder to container cache and then tidy or it seems the tidy didn't read the vendor folder. maybe we should set go mod to read vendor folder?

Reply:
it seems that https://github.com/golang/go/blob/15fa7a84b88165092d3a05fb0af11f11d967065d/src/cmd/go/internal/modcmd/tidy.go#L116 the `go mod tidy` will always use go mod without load go vendor, which makes GO111MODULE inefficacy.
Will further investigate and keep posted.